servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/ClosableConcurrentStack.java

```diff
@@ -70,15 +70,27 @@ boolean relaxedRemove(T item) {
             return false;
         }
         @SuppressWarnings("unchecked")
-        Node<T> currTop = (Node<T>) rawCurrTop;
-        while (currTop != null) {
-            if (item.equals(currTop.item)) {
-                currTop.item = null; // best effort null out the item. pop/close will discard the Node later.
+        Node<T> curr = (Node<T>) rawCurrTop;
+        Node<T> prev = null;
+        do {
+            if (item.equals(curr.item)) {
+                // Best effort Node removal. close will discard the Node if this attempt isn't visible to all threads.
+                curr.item = null;
+                if (prev != null) {
+                    prev.next = curr.next;
+                    // Don't set curr.next to null! If multiple threads are removing items at the same time removed
```

Comment:
this case is already covered by existing unit tests `ClosableConcurrentStackTest.concurrentPushRemoveDifferentThread`